### PR TITLE
Add old studio layout addon

### LINKED ISSUE #2935: 2.0-ify studios
### Is your request related to a problem? Please describe.

In the new studio theme the number of followers is missing

### Describe the solution you'd like

Show the number of followers In the new studio theme

### Additional context

none

### PR DESCRIPTION
**Pull request by mxmou**
Resolves #2935

### Changes

Adds an addon that reverts studios to either the old design or the 2.0 → 3.0 design. Adjusts studio-tools for compatibility with the added addon and attempts to fix a bug causing the leave button to not always appear.

### Screenshots

Default:
image

Scratch 2.0 → 3.0:
image

Adding projects:
image

### CODE REVIEW

**Comment by Scratchfangs** [comment]
Well, I'm excited for this

**Comment by CubeyTheCube** [comment]
Will there be an addon to change the header to 2.0 style??

**Comment by mxmou** [comment]
> Will there be an addon to change the header to 2.0 style??

That's something I want to do in the future, but probably not soon.

**Comment by WorldLanguages** [comment]
Oh wow. Definitely targeting for v1.17.0 in a few days, no matter what.

**Review by WorldLanguages — APPROVED** [review]
Waiting for @apple502j's review

**Comment by WorldLanguages** [comment]
Can anything be done about the size of the follow button? It creates a lot of empty space:
image

File: addons/old-studio-layout/addon.json [hunk 1]
```diff
@@ -0,0 +1,70 @@
+{
+  "name": "Old studio layout",
+  "description": "Reverts the layout of the studio page back to the one used before the studio update or the one previously available as part of the Scratch 2.0 \u2192 3.0 addon.",
+  "info": [
```
Comment (WorldLanguages) [inline addons/old-studio-layout/addon.json:4]: You might want to credit yourself

**Comment by mxmou** [comment]
> Can anything be done about the size of the follow button? It creates a lot of empty space:
> image

I don't like the empty space either, but I didn't want to change the size of things too much.

**Comment by jeffalo** [comment]
> > Can anything be done about the size of the follow button? It creates a lot of empty space:
> > image
> 
> I don't like the empty space either, but I didn't want to change the size of things too much.

i think in this case, removing whitespace is more important than not changing the size of things

**Comment by mxmou** [comment]
> > > Can anything be done about the size of the follow button? It creates a lot of empty space:
> > 
> > I don't like the empty space either, but I didn't want to change the size of things too much.
> 
> i think in this case, removing whitespace is more important than not changing the size of things

What about this?
image

**Review by apple502j — APPROVED** [review]

**Comment by WorldLanguages** [comment]
@mxmou I like that.

**Comment by WorldLanguages** [comment]
I would also would like to see something like this, as a setting that's off by default:
Changes are:
- 4 columns in grid
- Smaller font size in project title
- Smaller username, maybe
- Smaller profile picture (probably needs to be centered for it to look properly)

image

It makes a huuge difference, you can see way more projects:
image

**Comment by WorldLanguages** [comment]
While testing this, encountered website dark mode bug:

element is `.studio-info-box.studio-info-box-error` and it inherits font color from body:
image

**Comment by WorldLanguages** [comment]
@mxmou Please explicitly tell when this should be merged. I'd like to see 4 project rows as a setting, would be cool

**Comment by mxmou** [comment]
> @mxmou Please explicitly tell when this should be merged. I'd like to see 4 project rows as a setting, would be cool

It would be, but because this is a highly requested feature I think that could be added later.

**Comment by Secret-chest** [comment]
> > > > Can anything be done about the size of the follow button? It creates a lot of empty space:
> > > 
> > > I don't like the empty space either, but I didn't want to change the size of things too much.
> > 
> > i think in this case, removing whitespace is more important than not changing the size of things
> 
> What about this?
> image

@mxmou That looks bad. Add it as a setting, please. Your current one or Follow (506)

**Comment by CodeGuy92** [comment]
Capture
Bruh what

**Comment by WorldLanguages** [comment]
@CodeGuy92 how 😂

**Comment by WorldLanguages** [comment]
@mxmou Status on 1) smaller projects 4 per row and 2) dark mode bug? Both in comments above.

**Comment by mxmou** [comment]
> @mxmou Status on 1) smaller projects 4 per row and 2) dark mode bug? Both in comments above.

I'm working on a separate addon that will be able to change the numbers of projects per row on all pages that present them in a grid layout (Expore, search, user's shared projects, ...).

I don't think dark mode has much better supports for other types of errors - I will probably work on those in the future.

**Comment by CodeGuy92** [comment]
> @CodeGuy92 how 😂

idk that's just how it loaded lol

**Pull request merged**